ui/src/Menu.js

```diff
@@ -0,0 +1,102 @@
+import Typography from '@material-ui/core/Typography';
+import Divider from '@material-ui/core/Divider';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import Layout from './pageLayout.js';
+import CodeIcon from '@material-ui/icons/Code';
+import HomeIcon from '@material-ui/icons/Home';
+import TimelineIcon from '@material-ui/icons/Timeline';
+
+const drawerWidth = 240;
+const appBarHeight = 0;
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        display: 'flex',
+    },
+    appBar: {
+        width: `calc(100% - ${drawerWidth}px)`,
+        Height: appBarHeight,
+        marginLeft: drawerWidth,
+        marginBottom: 65,
+
+    },
+    drawer: {
+        width: drawerWidth,
+        flexShrink: 0,
+    },
+    drawerPaper: {
+        width: drawerWidth,
+    },
+    // necessary for content to be below app bar
+    toolbar: theme.mixins.toolbar,
+    content: {
+        flexGrow: 1,
+        backgroundColor: theme.palette.background.default,
+        padding: theme.spacing(3),
+
+    },
+    body: {
+        marginTop: appBarHeight,
+    }
+}));
+
+export default function PermanentDrawerLeft() {
+    const classes = useStyles();
+
+    return (
+        <div className={classes.root}>
+            <CssBaseline />
+            <AppBar position="fixed" className={classes.appBar} >
+                <Toolbar>
+                    <Typography variant="h6" noWrap>
+                        Challenge Name
+                    </Typography>
+                </Toolbar>
+            </AppBar>
+            <Drawer
+
+                className={classes.drawer}
+                variant="permanent"
+                classes={{
+                    paper: classes.drawerPaper,
+                }}
+                anchor="left"
+            >
+                <div className={classes.toolbar} />
+                <Divider />
+                <List>
+                    {['Home', 'Leaderboard'].map((text, index) => (
+                        <ListItem button key={text}>
+                            <ListItemIcon>{index % 2 === 0 ? <HomeIcon /> : <TimelineIcon />}</ListItemIcon>
+                            <ListItemText primary={text} />
+                        </ListItem>
+                    ))}
+                </List>
+                <Divider />
+                <List>
+                    <p align="center"></p>
+                    {['Challenge Category','Challenge 1', 'Challenge 2', 'Challenge 3'].map((text, index) => (
+                        <ListItem button key={text}>
+                            <ListItemIcon><CodeIcon/></ListItemIcon>
+                            <ListItemText primary={text} />
+                        </ListItem>
+                    ))}
+                </List>
+            </Drawer>
+            <main className={classes.content}>
+                <React.Fragment>
+                <Layout />
```

Comment:
I think we should have a component that adds the layout and the drawer, rather than the drawer adding the layout. This isn't a big deal right now though, we can change it later.

Reply:
I agree. I think its causing some issues with the sizing of the component windows.